source/slang/slang-emit-c-like.cpp

```diff
@@ -352,6 +352,50 @@ void CLikeSourceEmitter::_emitType(IRType* type, DeclaratorInfo* declarator)
     }
 }
 
+void CLikeSourceEmitter::_emitSwizzleStorePerElement(IRInst* inst)
+{
+    auto subscriptOuter = getInfo(EmitOp::General);
+    auto subscriptPrec = getInfo(EmitOp::Postfix);
+
+    // store the right-handside expression into a temporary variable first, if
+    // the right-handside expression is already a variable, we can skip this.
+    auto ii = cast<IRSwizzledStore>(inst);
+    String tempIdentifierName = "tmp";
```

Comment:
Don't do this yourself, instead, just make sure we don't fold `ii->getSource()` by handling a case for IRSwizzledStore use after line 1496 in slang-emit-c-like.cpp (inside CLikeSourceEmitter::shouldFoldInstIntoUseSites):

```
if (auto swizzledStore = as<IRSwizzledStore>(inst))
{
      if (isCPUTarget(getTargetReq()) || isCUDATarget(getTargetReq()))
           return false;
}
```

Reply:
This should make sure that we will always emit the source operand as a standalone assignment statement before we reach the swizzled store inst. So we can simply do a loop and call emitOperand(ii->getSource()) in the loop without duplicating the logic that computes `source`.